### PR TITLE
Simplifications in serializer

### PR DESCRIPTION
**Pull request by akva2**
Simplify the user interface for the serializer. All ops can now go through operator().
Downstream of https://github.com/OPM/opm-common/pull/3148
Sits on top of https://github.com/OPM/opm-simulators/pull/4107

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this opm-common=3148 please

File: ebos/eclmpiserializer.hh [hunk 1]
```diff
@@ -515,6 +242,174 @@ public:
...
+        } else {
+            if (m_op == Operation::UNPACK) {
+                std::size_t size = 0;
+                (*this)(size);
+                auto& data_mut = const_cast<std::vector<T>&>(data);
+                data_mut.resize(size);
+                std::for_each(data_mut.begin(), data_mut.end(), std::reference_wrapper{*this});
+            } else {
+                (*this)(data.size());
+                std::for_each(data.begin(), data.end(), std::reference_wrapper{*this});
```
Comment (bska) [inline ebos/eclmpiserializer.hh:274]: Tiny nit: You must `#include <functional>` to use `std::reference_wrapper`.  Once you include that header you can also just use the [`std::ref()`](https://en.cppreference.com/w/cpp/utility/functional/ref) helper function to make your intent even more explicit.

**Review by bska — COMMENTED** [review]
Looks good for the most part except for a small nit concerning `reference_wrapper`.  I like the simplification this work has brought.

File: ebos/eclmpiserializer.hh (hunk 1 above)
Comment (akva2) [reply ebos/eclmpiserializer.hh]: include added, std:.ref used. i will trigger the new build from upstream.

**Review by bska — APPROVED** [review]
Thanks a lot for the update.  I'll merge this into master now.

**Pull request merged**